### ISSUE TITLE
getSettingOrNull / getSettingOrDefault

### ISSUE THREAD

**Issue by filimonov** [ClickHouse/ClickHouse#68385]
Some setting profiles have custom setting, while other profiles - don't.
Need to access that in queries without exceptions.

**Comment by aadant** [comment]
@alexey-milovidov @tbragin @ArctypeZach ^ FYI

**Comment by shiyer7474** [comment]
PR for introducing `getSettingOrNull() `is ready, will upload shortly for review.

```
$ select getSetting('custom_xyz');

SELECT getSetting('custom_xyz')

Received exception from server (version 24.9.1):
Code: 115. DB::Exception: Received from localhost:9000. DB::Exception: Unknown setting 'custom_xyz': In scope SELECT getSetting('custom_xyz'). (UNKNOWN_SETTING)

$ select getSettingOrNull('custom_xyz');

   ┌─getSettingOrNull('custom_xyz')─┐
1. │ ᴺᵁᴸᴸ                           │
   └────────────────────────────────┘

1 row in set. Elapsed: 0.005 sec.

$ select * from t2;

   ┌─id─┐
1. │  1 │
2. │  2 │
   └────┘

2 rows in set. Elapsed: 0.004 sec.

$ select * from t2 where getSetting('custom_xyz') = 'Y';

Received exception from server (version 24.9.1):
Code: 115. DB::Exception: Received from localhost:9000. DB::Exception: Unknown setting 'custom_xyz': In scope SELECT * FROM t2 WHERE getSetting('custom_xyz') = 'Y'. (UNKNOWN_SETTING)

$ select * from t2 where getSettingOrNull('custom_xyz') = 'Y';

0 rows in set. Elapsed: 0.005 sec.

$ select * from t2 where getSettingOrNull('custom_xyz') = 'Y' or isNull(getSettingOrNull('custom_xyz'));

   ┌─id─┐
1. │  1 │
2. │  2 │
   └────┘

2 rows in set. Elapsed: 0.004 sec.
```

**Comment by shiyer7474** [comment]
A bit of rework done and `getSettingOrNull() `is replaced by `getSettingOrDefault()`

```
$ SELECT getSetting('custom_xyz') AS v

Received exception from server (version 24.9.1):
Code: 115. DB::Exception: Received from localhost:9000. DB::Exception: Unknown setting 'custom_xyz': In scope SELECT getSetting('custom_xyz'). (UNKNOWN_SETTING)


$ SELECT getSettingOrDefault('custom_xyz', 'my_default_value') AS v


   ┌─v────────────────┐
1. │ my_default_value │
   └──────────────────┘


$ SELECT getSettingOrDefault('custom_xyz', NULL) AS v

   ┌─v────┐
1. │ ᴺᵁᴸᴸ │
   └──────┘

$ SELECT getSettingOrDefault('custom_xyz', 50) AS v

   ┌──v─┐
1. │ 50 │
   └────┘

$ SELECT getSettingOrDefault(1, 'def') AS v

Code: 43. DB::Exception: Received from localhost:9000. DB::Exception: The argument of function getSettingOrDefault should be a constant string with the name of a setting: In scope SELECT getSettingOrDefault(1, 'def') AS v. (ILLEGAL_TYPE_OF_ARGUMENT)

$  SELECT count(*) FROM numbers(10) WHERE number = getSettingOrDefault('custom_role', 5)

   ┌─count()─┐
1. │       1 │
   └─────────┘

$ SELECT getSettingOrDefault('custom_xyz', NULL) IS NULL AS v

   ┌─v─┐
1. │ 1 │
   └───┘


$ SET custom_run_role = 'role:primary_1'

Ok.

$ SELECT getSettingOrDefault('custom_run_role', 'role:none')


   ┌─getSettingOrDefault('custom_run_role', 'role:none')─┐
1. │ role:primary_1                                      │
   └─────────────────────────────────────────────────────┘

```

**Comment by alexkats** [comment]
Fixed by #69917
